src/openfermion/utils/_sparse_tools.py

```diff
@@ -203,6 +203,71 @@ def qubit_operator_sparse(qubit_operator, n_qubits=None):
     sparse_operator.eliminate_zeros()
     return sparse_operator
 
+def get_linear_qubit_operator(qubit_operator, n_qubits=None):
+    """ Return a linear operator with matvec defined to avoid instantiating a
+    huge matrix, which requires lots of memory.
+
+    Args:
+      qubit_operator(QubitOperator): A qubit operator to be applied on vectors.
+
+    Returns:
+      linear_operator(LinearOperator): The equivalent operator which is well
+      defined when applying to a vector.
+
+    """
+    if n_qubits is None:
+        n_qubits = count_qubits(qubit_operator)
+    if n_qubits < count_qubits(qubit_operator):
+        raise ValueError('Invalid number of qubits specified.')
+
+    n_hilbert = 2 ** n_qubits
+
+    def matvec(vec):
+        if (len(vec) != n_hilbert):
```

Comment:
My understand is that len(vec) could be larger than n_hilbert. As long as len(vec) equals to 2**m (where m >= n_qubits), it is OK. We can simply think "X0 Y1 Z4" as "X0 Y1 I2 I3 Z4 I5 ..."  (where I is identity op)?
@babbush @jarrodmcc